src/sa_web/static/js/routes.js

```diff
@@ -58,32 +67,53 @@ var Shareabouts = Shareabouts || {};
         }
       }, this);
 
-      this.loading = true;
-      this.collection = new S.PlaceCollection([]);
-      this.activities = new S.ActionCollection(options.activity);
+      this.loading = true;      
 
+      var configArrays = {};
 
-      this.landmarkCollections = {};
-      var landmarkConfigsArray = options.mapConfig.layers.filter(function(layer) {
+      configArrays.landmark = options.mapConfig.layers.filter(function(layer) {
         return layer.type && layer.type === 'landmark';
       });
-      var landmarkConfigs = {};
-      _.each(landmarkConfigsArray, function(landmarkConfig) {
-        var collectionId = landmarkConfig['id'];
-        var collection = new S.LandmarkCollection([]);
-        collection.url = landmarkConfig.url;
-        self.landmarkCollections[collectionId] = collection;
-        landmarkConfigs[collectionId] = landmarkConfig;
+      configArrays.place = options.mapConfig.layers.filter(function(layer) {
+        return layer.type && layer.type === 'place';
+      });
+     
+      _.each(configArrays, function(configArray) {
+        _.each(configArray, function(config) {
+          var collection;
+          if (config.type == "place") {
+            collection = new S.ActionCollection([]);
+            collection.url = "/dataset/" + config.id + "/actions";
+            self.activities[config.type][config.id] = collection;
+          }
+        });
+      });
+
+      _.each(configArrays, function(configArray) {
+        _.each(configArray, function(config) {
+          var collection;
+          if (config.type == "place") {
+            collection = new S.PlaceCollection([]);
+            collection.url = "/dataset/" + config.id + "/places";
+          } else if (config.type == "landmark") {
+            collection = new S.LandmarkCollection([]);
+            collection.url = config.url;
+          }
```

Comment:
We might want to explore simplifying these double for-loops again. 

Of course, we can always do it after merging.